tools/istio-iptables/pkg/capture/run.go

```diff
@@ -862,51 +864,57 @@ check_loop:
 		output, err := cfg.ext.RunWithOutput(constants.IPTablesSave, ipCfg.ver, nil)
 		if err == nil {
 			currentState := cfg.getStateFromSave(output.String())
+			log.Debugf("Current iptables state: %s", currentState)
 			for _, value := range currentState {
-				if residueFound {
+				if deltaExists {
 					break
 				}
-				residueFound = len(value) != 0
+				deltaExists = len(value) != 0
 			}
-			if !residueFound {
+			if !deltaExists {
 				continue
 			}
 			expectedState := cfg.getStateFromSave(ipCfg.expected)
+			log.Debugf("Expected iptables state: %s", expectedState)
 			for table, chains := range expectedState {
 				_, ok := currentState[table]
 				if !ok {
-					applyRequired = true
+					reconcile = true
+					log.Debugf("Can't find expected table %s in current state", table)
 					break check_loop
 				}
 				for chain, rules := range chains {
 					currentRules, ok := currentState[table][chain]
 					if !ok || (strings.HasPrefix(chain, "ISTIO_") && len(rules) != len(currentRules)) {
-						applyRequired = true
+						reconcile = true
+						log.Debugf("Mismatching number of rules in chain %s between current and expected state", chain)
 						break check_loop
 					}
 				}
 			}
 			err = cfg.executeIptablesCommands(ipCfg.ver, ipCfg.checkRules)
 			if err != nil {
-				applyRequired = true
+				reconcile = true
+				log.Debugf("iptables check rules failed")
 				break
 			}
 		}
 
 	}
 
-	// If there are no residue, apply step is always needed
-	if !residueFound {
-		return false, true
+	// If no delta is found, reconcile is never needed
+	if !deltaExists {
+		log.Info("Clean-state detected, no reconciliation needed")
+		return false, false
 	}
 
-	if residueFound && !applyRequired {
-		log.Info("Found compatible iptables rules/chains, no additional changes are needed")
-	} else if residueFound {
-		log.Info("Found residue of old iptables rules/chains, cleanup is needed")
+	if deltaExists && !reconcile {
+		log.Info("Found compatible iptables rules/chains, no reconciliation needed")
```

Comment:
This log line doesn't seem to be correct anymore, should be "Found residue, reconciliation disabled"?

Reply:
My bad here, the reconcile variable was not supposed to be mentioned in VerifyIptablesState().
Code has been adjusted now to mention residueExists and deltaExists

Fast-fail based on reconcile enabled/disabled is here: https://github.com/istio/istio/blob/cad8700a8e6d6cddcdb47eaf0ca61767e045c8af/tools/istio-iptables/pkg/capture/run.go#L931-L933